functions/check_deploy.js

```diff
@@ -1,56 +1,48 @@
 "use strict";
 
-const http = require("http");
-const url = require("url");
+const axios = require("axios");
+const serverUtils = require("../functions-common/server_utils.js");
+
+/**
+ * Support the endpoint /api/check_deploy
+ */
 
 exports.handler = async function (event) {
     console.log("Received check_deploy request at " + new Date() + " with path " + event.path);
-    var branch = /check_deploy\/(.*)/.exec(event.path)[1];
+    const incomingData = JSON.parse(event.body || {});
 
     // Reject the request when:
-    // 1. Not a GET request;
+    // 1. Not a POST request;
     // 2. Doesn’t provide required values
-    if (event.httpMethod !== "GET" || !branch ||
-        !process.env.ACCESS_TOKEN || !process.env.WORDLES_REPO_OWNER || !process.env.WORDLES_REPO_NAME) {
-        return {
-            statusCode: 400,
-            body: JSON.stringify({
-                message: "Invalid HTTP request method or missing field values or missing environment variables."
-            })
-        };
+    if (event.httpMethod !== "POST" || !serverUtils.isParamsExist([incomingData.branches])) {
+        return serverUtils.invalidRequestResponse;
     }
 
-    const checkUrlExists = async function (wordleUrl) {
-        return new Promise(function (resolve, reject) {
-            const req = http.request({
-                method: "HEAD",
-                host: url.parse(wordleUrl).host,
-                port: 80,
-                path: url.parse(wordleUrl).pathname
-            }, function (response) {
-                if (response.statusCode < 400) {
-                    resolve(true);
-                } else {
-                    resolve(false);
-                }
-            });
-
-            req.on("error", err => {
-                reject(err);
-            });
-            req.end();
-        });
+    /**
+     * Check if an URL exists.
+     * @param {String} url - The URL to check.
+     * @return {Boolean} Return true if the URL exists. Otherwise, return false.
+     */
+    const urlExists = async function (url) {
+        try {
+            // When the url doesn't exist, the axios call will fail with 404 status and fall into the catch block.
+            await axios.head(url);
+            return true;
+        } catch (e) {
+            return false;
+        }
     };
 
     try {
-        const wordleUrl = "https://" + branch + "--inverted-wordles.netlify.app/";
-        const isUrlExists = await checkUrlExists(wordleUrl);
-        console.log("Done: the wordle url existent status", isUrlExists);
+        let resultsTogo = {};
+        for (const branch of incomingData.branches) {
+            const wordleUrl = "https://" + branch + "--inverted-wordles.netlify.app/";
```

Comment:
This URL can't be depended on to be stable. For my personal deploy of the app, it was deployed to https://zealous-mayer-52778b.netlify.app/ - and for example I have made a branch at https://github.com/amb26/inverted-wordles/tree/936975c8-b8af-49e0-8a6e-f4be1440fb4f which does not appear at https://936975c8-b8af-49e0-8a6e-f4be1440fb4f--inverted-wordles.netlify.app/

Probably the API will have to be improved to have the client supply its own URL in the payload? But I'm not sure this can be depended on since someday the site could be hosted at a different URL. I think a stable solution to this will require more digging into the output of the "deploys" API - this will also allow us to display more helpful status about failed deploys before the 2 minute timeout.

Reply:
I did look into [the netlify deploy API](https://docs.netlify.com/api/get-started/#deploy-via-api). What I learnt is, when a deploy is manually triggered via this API, a deploy id will be returned, which can be used to poll this API with to find out the deploy status. However, two methods to trigger a deploy via API is either by uploading files or a zipped package of files to this API. In our case, the deploy is automatically triggered when a new Github branch is created. I haven't found a Netlify API endpoint that returns the deploy id when a deploy is automatically triggered.

Another solution is, we move from the automatic branch deploy to a manual API deploy. However, this change will cause the change to the entire workflow that we cannot list all wordle sites by browsing through Github branches but by querying via the netlify API.

As to the url of a deploy, you are right that looking up `--inverted-wordles.netlify.app/` is improper when you have personal custom netlify domain. The domain of `inverted-wordles` is the domain defined for the production site. BTW, for branch deploys, the format of `branchname--yoursitename.netlify.app` is valid according to [the netlify documentation](https://docs.netlify.com/site-deploys/overview/#definitions). A possible solution for this is to define an extra environment variable for `yoursitename`.

Please let me know about your thoughts. Thanks, @amb26.

Reply:
Rather than the "deploy API" itself, I rather meant to query the deploys API endpoint at https://open-api.netlify.com/#tag/deploy to see if we can locate what deploys have been done recently or are in progress, and associate them with their respective github branches.

It appears that netlify defines sufficient default environment variables to allow the site to locate itself listed at https://docs.netlify.com/configure-builds/environment-variables/ so we can probably eliminate the need to locate REPOSITORY_URL etc. - we should adjust our app so that it consumes netlify's default variables rather than WORDLES_REPO_OWNER, WORDLES_REPO_NAME etc. so that someone setting up a new site has as little work to do as possible.

Reply:
@amb26, after some experiments, I found REPOSITORY_URL is only available with actual Netlify sites but unavailable with local dev runs using `netlify dev` command.

A workaround is to update the documentation to indicate `REPOSITORY_URL` needs to be manually defined as an environment variable when running locally but unnecessary with netlify.com. What do you think?

Reply:
That seems fine to me and what I expected. I think the win is helpful in reducing one piece of bureaucracy for people running real sites.

Reply:
Great. Will make the change.